pkgs/by-name/ze/zenmap/package.nix

```
@@ -0,0 +1,76 @@
+{
+  gobject-introspection,
+  gtk3,
+  lib,
+  nmap,
+  python3Packages,
+  wrapGAppsHook3,
+  xterm,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "zenmap";
+  version = nmap.version;
+  pyproject = true;
+
+  src = nmap.src;
+
+  sourceRoot = "${nmap.name}/zenmap";
+
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  buildInputs = [
+    nmap
+    gtk3
+    xterm
+  ];
+
+  nativeBuildInputs = [
+    wrapGAppsHook3
+    gobject-introspection
+  ];
+
+  dependencies = with python3Packages; [
+    pygobject3
+  ];
+
+  postInstall = ''
+    wrapProgram "$out/bin/zenmap" \
+      --prefix PATH : ${lib.makeBinPath [ nmap ]}
+    # Icons
+    install -Dm 644 "zenmapCore/data/pixmaps/zenmap.png" -t "$out/share/pixmaps/"
+    # Desktop-files for application
+    install -Dm 644 "install_scripts/unix/zenmap.desktop" -t "$out/share/applications/"
+    substituteInPlace "$out/share/applications/zenmap.desktop" \
+        --replace-fail "Exec=zenmap %F" \
+                      "Exec=$out/bin/zenmap %F"
+    install -Dm 644 "install_scripts/unix/zenmap-root.desktop" -t "$out/share/applications/"
+    substituteInPlace "$out/share/applications/zenmap-root.desktop" \
+        --replace-fail "TryExec=su-to-zenmap.sh" \
+                      "TryExec=$out/bin/su-to-zenmap.sh" \
+        --replace-fail "Exec=su-to-zenmap.sh %F" \
+                      "Exec=$out/bin/su-to-zenmap.sh %F"
+    install -Dm 755 "install_scripts/unix/su-to-zenmap.sh" -t "$out/bin/"
+    substituteInPlace "$out/bin/su-to-zenmap.sh" \
+        --replace-fail 'COMMAND="zenmap"' \
+                      'COMMAND="'"$out/bin/zenmap"'"' \
+        --replace-fail 'xterm' \
+                      '"${xterm}/bin/xterm"'
+  '';
+
+  checkPhase = ''
+    cd test
+    ${python3Packages.python.interpreter} run_tests.py
+  '';
```

```suggestion
  checkPhase = ''
    runHook preCheck

    cd test
    ${python3Packages.python.interpreter} run_tests.py 2>&1 | tee /dev/stderr | tail -n1 | grep '^OK$'

    runHook postCheck
  '';
```
The check script always exits with code 0, so we must check for passing tests.
When overriding a phase the pre and post hooks must be run.